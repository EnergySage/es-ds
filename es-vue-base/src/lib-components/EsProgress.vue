<template>
    <b-progress
        aria-describedby="es-progress"
        class="EsProgress bg-gray-200 rounded-0"
        :class="{ 'EsProgress--full': value === 100 }"
        :height="height"
        :value="value"
        v-bind="$attrs"
        v-on="$listeners" />
</template>

<script lang="js">
import { BProgress } from 'bootstrap-vue';

export default {
    name: 'EsProgress',
    components: {
        BProgress,
    },
    props: {
        height: {
            type: String,
            default: '0.3125rem', // 5px
        },
        value: {
            type: Number,
            required: true,
        },
    },
};
</script>

<style lang="scss" scoped>
.EsProgress {
  ::v-deep .progress-bar {
    border-radius: 0 0.125rem 0.125rem 0;
  }
}

.EsProgress--full {
  ::v-deep .progress-bar {
    border-radius: 0;
  }
}
</style>
