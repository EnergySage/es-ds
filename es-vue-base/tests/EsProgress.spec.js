import { mount } from '@vue/test-utils';
import EsProgress from '@/src/lib-components/EsProgress.vue';
import jestVue from '@/tests/jest.vue.config';

describe('EsProgress', () => {
    test('<EsProgress />', async () => {
        const wrapper = mount(EsProgress, {
            ...jestVue,
            propsData: {
                value: 55,
            },
        });

        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('check for <EsProgress />', async () => {
        const wrapper = mount(EsProgress, {
            ...jestVue,
            propsData: {
                value: 55,
            },
        });

        expect(wrapper.find('.progress').exists()).toBe(true);
    });

    // test default prop values
    test('<EsProgress /> default prop values', async () => {
        const wrapper = mount(EsProgress, {
            ...jestVue,
            propsData: {
                value: 55,
            },
        });
        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();

        expect(wrapper.props('circle')).toBe(false);
        expect(wrapper.props('showPercentage')).toBe(true);
    });

    // test circular variant
    test('<EsProgress /> circular variant', async () => {
        const wrapper = mount(EsProgress, {
            ...jestVue,
            propsData: {
                circle: true,
                showPercentage: false,
                value: 21,
            },
        });
        expect(wrapper.vm).toBeTruthy();
        expect(wrapper.html()).toMatchSnapshot();

        expect(wrapper.props('circle')).toBe(true);
        expect(wrapper.props('showPercentage')).toBe(false);
    });
});
