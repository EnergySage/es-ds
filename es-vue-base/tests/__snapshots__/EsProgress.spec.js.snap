// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EsProgress <EsProgress /> 1`] = `
"<div class="progress EsProgress bg-gray-200 rounded-0" style="height: 0.3125rem;" aria-describedby="es-progress">
  <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="55" class="progress-bar" style="width: 55.00000000000001%;"></div>
</div>"
`;
